tests/optim/core/test_loss.py

```diff
@@ -0,0 +1,208 @@
+        self.assertAlmostEqual(get_loss_value(model, loss), cos_sim, places=6)
+
+
+class TestTensorDirection(BaseTest):
+    def test_tensor_direction(self) -> None:
+        model = SimpleModel()
+        loss = opt_loss.TensorDirection(model.layer, vec=torch.ones(1, 1, 1, 1))
+        a = 1
+        b = [CHANNEL_ACTIVATION_0_LOSS, CHANNEL_ACTIVATION_1_LOSS]
+        cos_sim = np.inner(a, b) / (np.linalg.norm(a) * np.linalg.norm(b))
+        cos_sim = np.sum(cos_sim)
+        self.assertAlmostEqual(get_loss_value(model, loss), cos_sim, places=6)
+
+
+class TestActivationWeights(BaseTest):
+    def test_activation_weights_0(self) -> None:
+        model = SimpleModel()
+        loss = opt_loss.ActivationWeights(model.layer, weights=torch.zeros(1))
+        assertArraysAlmostEqual(get_loss_value(model, loss), np.zeros((1, 2, 1)))
+
+    def test_activation_weights_1(self) -> None:
+        model = SimpleModel()
+        loss = opt_loss.ActivationWeights(
+            model.layer, weights=torch.ones(1), neuron=True
+        )
+        assertArraysAlmostEqual(
+            get_loss_value(model, loss),
+            [CHANNEL_ACTIVATION_0_LOSS, CHANNEL_ACTIVATION_1_LOSS],
+        )
+
+
+class TestCompositeLoss(BaseTest):
+    def test_negative(self) -> None:
+        model = SimpleModel()
+        loss = -opt_loss.ChannelActivation(model.layer, 0)
+        self.assertAlmostEqual(get_loss_value(model, loss), -CHANNEL_ACTIVATION_0_LOSS)
+
+    def test_addition(self) -> None:
+        model = SimpleModel()
+        loss = (
+            opt_loss.ChannelActivation(model.layer, 0)
+            + opt_loss.ChannelActivation(model.layer, 1)
+            + 1
+        )
+        self.assertAlmostEqual(
+            get_loss_value(model, loss),
+            CHANNEL_ACTIVATION_0_LOSS + CHANNEL_ACTIVATION_1_LOSS + 1,
+        )
+
+    def test_subtraction(self) -> None:
+        model = SimpleModel()
+        loss = (
+            opt_loss.ChannelActivation(model.layer, 0)
+            - opt_loss.ChannelActivation(model.layer, 1)
+            - 1
+        )
+        self.assertAlmostEqual(
+            get_loss_value(model, loss),
+            CHANNEL_ACTIVATION_0_LOSS - CHANNEL_ACTIVATION_1_LOSS - 1,
+        )
+
+    def test_multiplication(self) -> None:
+        model = SimpleModel()
+        loss = opt_loss.ChannelActivation(model.layer, 0) * 10
+        self.assertAlmostEqual(
+            get_loss_value(model, loss), CHANNEL_ACTIVATION_0_LOSS * 10
+        )
+
+    def test_division(self) -> None:
+        model = SimpleModel()
+        loss = opt_loss.ChannelActivation(model.layer, 0) / 10
+        self.assertAlmostEqual(
+            get_loss_value(model, loss), CHANNEL_ACTIVATION_0_LOSS / 10
+        )
+
+    def test_pow(self) -> None:
+        model = SimpleModel()
+        loss = opt_loss.ChannelActivation(model.layer, 0) ** 2
+        self.assertAlmostEqual(
+            get_loss_value(model, loss), CHANNEL_ACTIVATION_0_LOSS ** 2
+        )
+
```

Comment:
Good test cases! In order to test the edge cases with CompositeLoss you can also pass non (int, float) and catch the case:
https://github.com/pytorch/captum/pull/552/files#diff-9a1b59d73279e5933b25a842cf13915482ca23707ef4ccf4d522b6e66edde8c6R64
e.g
You could assert it `with assertRaises`
https://github.com/pytorch/captum/blob/master/tests/attr/test_deeplift_basic.py#L283

Reply:
Thanks! Yup I'll add these tests in a later commit!

Reply:
Added test cases!